### PR TITLE
Reenables taking info tags from corpses

### LINKED ISSUE #13806: trying to place ID cards and dogtags on the shipside memorial no longer works
### Description

i hoped to be able to take a tag off a dead marine and place it on the memorial on the ship so that it would add their name to the memorial

no interaction happens and the tag stays in your hand. the memorial doesn't update or anything.

this has been broken for a while; at least since early june.

### Test Merges

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/sybil/server_side_modifications.dm

Round ID: [22799](https://scrubby.melonmesa.com/round/22799)
BYOND Version: 515.1610
Server revision compiled on: 2023-08-15T08:03:29+00:00
Master commit: [5e28c93215e15791b89f57a3176a78a55a6e3c13](https://github.com/tgstation/TerraGov-Marine-Corps//commit/5e28c93215e15791b89f57a3176a78a55a6e3c13)
The following pull requests are currently test merged:
[#13515: 'Adds 2 new tadpole models updating: to make grief poles less griefy' by XANAv2 at commit e52d05bff0](https://github.com/tgstation/TerraGov-Marine-Corps//pull/13515)
[#13308: 'New Xenomorph : Baneling' by Zar000 at commit c1587b8f60](https://github.com/tgstation/TerraGov-Marine-Corps//pull/13308)
[#13626: 'Slumbridge' by chizzy376 at commit 43748212ff](https://github.com/tgstation/TerraGov-Marine-Corps//pull/13626)
[#13767: 'Scout ammo changes' by Lumipharon at commit 97965d4a3c](https://github.com/tgstation/TerraGov-Marine-Corps//pull/13767)
[#13799: 'Probably fixes AI biospam' by RipGrayson at commit 04a592b595](https://github.com/tgstation/TerraGov-Marine-Corps//pull/13799)

Local commit: 13e1c9499b255dcb94f6db968ef03994c64786fd
Server tools version: 5.13.4

### Reproduction Steps

1. find permadead marine
2. remove their tag
3. take it to the shipside memorial slab
4. click the slab with the tag
5. nothing happens

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
Previously when you attempted to take a dogtag ID, you instead generated a separate item, the info tag, which only held name and job and could stack with multiple other info tags and be put in the ship memorial. All that got removed by the strip UI overhaul, and so this PR puts it back in.

Adds routing to enable items to react to being stripped, like a gun exploding if someone hostile takes it or vital intelligence sending out a warning signal when it's stolen, plus more variable feedback options for why you can't strip something.
## Why It's Good For The Game
Fixes #13806 
Also a couple minor codeside spelling things.
## Changelog
:cl:
fix: Taking info tags from corpses works again
/:cl:

### CODE REVIEW

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -676,8 +676,16 @@ GLOBAL_DATUM_INIT(welding_sparks_prepdoor, /mutable_appearance, mutable_appearan
...
 /obj/item/proc/canStrip(mob/stripper, mob/owner)
-	return !(flags_item & NODROP)
+	if(flags_item & NODROP)
+		stripper.balloon_alert(stripper, "[src] is stuck!")
+		return FALSE
+	return TRUE
+
+/// Used by any item which wants to react to or prevent its own stripping, called after checks/delays. Return TRUE to block normal stripping behavior.
+/obj/item/proc/special_stripped_behavior(mob/stripper, mob/owner)
+
```
Comment (TiviPlus) [inline code/game/objects/items.dm:688]: add a
tab return

stub

File: code/game/objects/items/cards_ids.dm [hunk 2]
```diff
@@ -250,6 +250,29 @@
...
+		stripper.balloon_alert(stripper, "Info tag already taken")
+		return FALSE
+	if(owner.stat != DEAD)
+		stripper.balloon_alert(stripper, "[owner] isn't dead yet")
+		return FALSE
+
+/obj/item/card/id/dogtag/special_stripped_behavior(mob/stripper, mob/owner)
+	if(dogtag_taken)
+		return
+	to_chat(stripper, span_notice("You take [owner]'s information tag, leaving the ID tag"))
```
Comment (TiviPlus) [inline code/game/objects/items/cards_ids.dm:267]: balloon?

File: code/game/objects/items/cards_ids.dm [hunk 3]
```diff
@@ -250,6 +250,29 @@
...
+	dogtag_taken = TRUE
+	icon_state = "dogtag_taken"
```
Comment (TiviPlus) [inline code/game/objects/items/cards_ids.dm:269]: gross why not updateicon

File: code/game/objects/items/cards_ids.dm (hunk 2 above)
Comment (QualityVan) [reply code/game/objects/items/cards_ids.dm]: I wanted the extra info which'd be too wordy for just a balloon, but I've added a shorter one on top now.

File: code/game/objects/items/cards_ids.dm (hunk 3 above)
Comment (QualityVan) [reply code/game/objects/items/cards_ids.dm]: the allure of copypasting old code and thinkn't

File: code/game/objects/items/cards_ids.dm [hunk 4]
```diff
@@ -250,6 +250,36 @@
 	iff_signal = TGMC_LOYALIST_IFF
 	var/dogtag_taken = FALSE
 
+/obj/item/card/id/dogtag/update_icon_state()
+	if(dogtag_taken)
+		icon_state = initial(icon_state) + "_taken"
+		return
+	icon = initial(icon_state)
```
Comment (TiviPlus) [inline code/game/objects/items/cards_ids.dm:257]:
```suggestion
	icon_state = initial(icon_state)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm
